sdk/go/x/auto/local_workspace_test.go

```diff
@@ -32,6 +33,71 @@ import (
 
 const pulumiOrg = "pulumi"
 
+func TestWorkspaceSecretsProvider(t *testing.T) {
+	ctx := context.Background()
+	pName := "go_secrets_provider"
+	sName := fmt.Sprintf("int_test%d", rangeIn(10000000, 99999999))
+	fqsn := FullyQualifiedStackName(pulumiOrg, pName, sName)
+
+	// We can't use Workspace EnvVars as the Workspace uses the secrets provider to
+	// create the Stack
+	os.Setenv("PULUMI_CONFIG_PASSPHRASE", "password")
```

Comment:
Note that you can do something along the lines of: 

```go
w, err := NewLocalWorkspace( Program(...), SecretsProvider(...), Project(...))
w.SetEnv()
NewStack(ctx, fqsn, ...)
```

This is fine for now. CC @metral what do you think about adding environment variables as workspace options to make this a little more ergonomic? That way both SecretsProvider and Envs could be specified at workspace creation time and enable using the `NewStackInlineSource` helper?

Reply:
> what do you think about adding environment variables as workspace options to make this a little more ergonomic? That way both SecretsProvider and Envs could be specified at workspace creation time

SGTM. Feel free to open an issue and assign it to me